src/pybind/mgr/cephadm/module.py

```diff
@@ -2859,6 +2861,38 @@ def set_prometheus_access_info(self, user: str, password: str) -> str:
         self.set_store(PrometheusService.USER_CFG_KEY, user)
         self.set_store(PrometheusService.PASS_CFG_KEY, password)
         return 'prometheus credentials updated correctly'
+    
+    @handle_orch_error
+    def set_prometheus_target(self, url: str) -> str:
+        prometheus_spec = cast(PrometheusSpec, self.spec_store['prometheus'].spec)
+        if url not in prometheus_spec.targets:
+            prometheus_spec.targets.append(url)
+        else:
+            raise OrchestratorError('Target {} already exists'.format(url))
+        if not prometheus_spec:
+            raise OrchestratorError('Service prometheus not found')
+        daemons: List[orchestrator.DaemonDescription] = self.cache.get_daemons_by_type('prometheus')
+        spec = ServiceSpec.from_json(prometheus_spec.to_json())
+        self.apply([spec], no_overwrite=False)
+        for daemon in daemons:
+            self.daemon_action(action='redeploy', daemon_name=daemon.daemon_name)
+        return 'prometheus multi-cluster targets updated'
+
+    @handle_orch_error
+    def remove_prometheus_target(self, url: str) -> str:
+        prometheus_spec = cast(PrometheusSpec, self.spec_store['prometheus'].spec)
+        if url in prometheus_spec.targets:
+            prometheus_spec.targets.remove(url)
+        else:
+            raise OrchestratorError('Target {} does not exist'.format(url))
```

Comment:
Maybe we should just show some error log here without raising an exception to make this command idempotent. Not sure though! probably @adk3798 could have a stronger opinion here.

Reply:
I'd say I lean towards a message saying the target already didn't exist but still a 0 return code for cases like this. So roughly what Redo is suggesting I think.